### PR TITLE
{bio,vis}[foss/2023a] SCENICplus v1.0a1-20240513, gensim v4.3.2, pyfasta v0.5.2, numexpr v2.9.0, Kaleido v0.2.1, Sphinx-RTD-Theme v2.0.0, scrublet v0.2.3

### PR DESCRIPTION
**Pull request by pavelToman**
(created using `eb --new-pr`)
resolves https://github.com/vscentrum/vsc-software-stack/issues/64

### CODE REVIEW

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by smoors** [comment]
> Test report by @boegelbot **FAILED** Build succeeded for 6 out of 8 (7 easyconfigs in total) cns1 - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) CPU E5-2667 v3 @ 3.20GHz (haswell), Python 3.6.8 See https://gist.github.com/boegelbot/de0b345903b78154ef118c9c68d9d858 for a full test report.

`polars` needs to be built from branch 5.0.x, i spawned a build for generoso in #20275

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

**Comment by pavelToman** [comment]
Seems `tensorflow-probability/0.20.0-foss-2023a` is not installed on generoso?

**Comment by smoors** [comment]
> Seems `tensorflow-probability/0.20.0-foss-2023a` is not installed on generoso?

it was rebuilt this morning, which probably clashed with your test
see https://github.com/easybuilders/easybuild-easyconfigs/pull/21172

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

**Comment by pavelToman** [comment]
@boegelbot please test @ jsc-zen3

File: easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb [hunk 1]
```diff
@@ -0,0 +1,281 @@
+easyblock = 'PythonBundle'
+
+name = 'scenicplus'
+version = '1.0a1-18072024'
```
Comment (boegel) [inline easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb:4]: datestamp should be `<year><month><day>`, and should correspond to date of commit being used (so in this case `20240513`)

File: easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb [hunk 2]
```diff
@@ -0,0 +1,281 @@
...
+        'checksums': ['7285ac03be9d148437442a726366d7c6f9f5f86bc6636369149a2bc2cd8e0e29'],
+    }),
+    (name, version, {
+        'preinstallopts': local_preinstallopts,
+        'source_urls': ['https://github.com/aertslab/scenicplus/archive/'],
+        'sources': [{'download_filename': 'fa55dae.tar.gz', 'filename': '%(name)s-%(version)s-fa55dae.tar.gz'}],
+        'checksums': [
+            {'scenicplus-1.0a1-18072024-fa55dae.tar.gz':
+             '88cddec1ab2618861e5c93e8a0b17b8e9e2aa3a76410d882c35d472f98724e29'},
+        ],
```
Comment (boegel) [inline easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb:269]:
```suggestion
        'sources': [{'download_filename': '%s.tar.gz' % local_commit, 'filename': SOURCE_TAR_GZ}],
        'checksums': ['88cddec1ab2618861e5c93e8a0b17b8e9e2aa3a76410d882c35d472f98724e29'],

File: easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    }),
+]
+
+# copy files to let 'scenicplus init_snakemake' works 
+postinstallcmds = [
+    'cp -r %(builddir)s/scenicplus/scenicplus-fa55dae55203951c0c42e359b65a28010cd544f6/src/scenicplus/snakemake '
```
Comment (boegel) [inline easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb:275]: don't hardcode commit ID here, use `local_commit` variable:
```suggestion
    "cp -r %%(builddir)s/scenicplus/scenicplus-%s*/src/scenicplus/snakemake " % local_commit
```

File: easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+# unpin version restriction and delete typing requirement
+local_pycistopic_preinstallopts = (
+    "sed -i"
+    " -e 's/pyscenic>=0.12.0/pyscenic/'"
+    " -e 's/pandas==1.5/pandas/'"
+    " -e 's/python-Levenshtein/Levenshtein/'"
+    " -e 's/python-igraph/igraph/'"
+    " -e '/typing/d'"
+    # delete pyscenic and loomxpy req from commit - have versions now
+    " -e '60,61d'"
```
Comment (boegel) [inline easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb:76]: Can we avoid relying on line numbers here, and use something like `/loomxpy/d` instead?

**Review by boegel — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb [hunk 5]
```diff
@@ -0,0 +1,281 @@
+easyblock = 'PythonBundle'
+
+name = 'scenicplus'
```
Comment (boegel) [inline easybuild/easyconfigs/s/scenicplus/scenicplus-1.0a1-18072024-foss-2023a.eb:3]: @pavelToman `SCENIC+` is official software name, so we should use that.

The `+` is annoying though, so let's go with `SCENISplus` as a compromise?

**Review by boegel — CHANGES_REQUESTED** [review]

**Comment by pavelToman** [comment]
@boegelbot please test @ generoso

File: easybuild/easyconfigs/s/SCENICplus/SCENICplus-1.0a1-20240513-foss-2023a.eb [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+        'checksums': ['82edc9313bbb587aeb1540fd5b095b6eafb422fa12a5fc97ba090b99b684b240'],
+    }),
+    ('pycistarget', '1.0a1-16d14b9', {
+        'preinstallopts': local_pycistarget_preinstallopts,
+        # download commit 16d14b9 to make '$ scenicplus' work
+        'source_urls': ['https://github.com/aertslab/pycistarget/archive/'],
+        'sources': [{'download_filename': '16d14b9.tar.gz', 'filename': '%(name)s-%(version)s-16d14b9.tar.gz'}],
+        'checksums': ['7285ac03be9d148437442a726366d7c6f9f5f86bc6636369149a2bc2cd8e0e29'],
+    }),
+    ('%(namelower)s', version, {
```
Comment (boegel) [inline easybuild/easyconfigs/s/SCENICplus/SCENICplus-1.0a1-20240513-foss-2023a.eb:263]: Avoid using `%(namelower)s` in extension name (there's an unresolved bug in EasyBuild framework that leads to not resolving this template):
```suggestion
    ('scenisplus', version, {

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 7 out of 7 (7 easyconfigs in total)
node3123.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/d8499e12f96e77a3c3d08a5291b7a3b4 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @pavelToman!

**Pull request merged**